docs/getting-started-on-snowplow-open-source/setup-snowplow-on-gcp/setup-google-cloud-storage-gcs-destination/index.md

```diff
@@ -4,10 +4,12 @@ date: "2020-03-02"
 sidebar_position: 60
 ---
 
-Snowplow supports streaming data (either raw or enriched data) from Pub/Sub into Google Cloud Storage (GCS). This can be a useful backup: it means that if there's an issue with the pipeline downstream (e.g. loading the data into BigQuery), there is the possibility of reloading the data from a GCS backup.
+Snowplow supports streaming data (either raw or enriched data) from [Pub/Sub](https://cloud.google.com/pubsub/) into [Google Cloud Storage (GCS)](https://cloud.google.com/storage/) with the [Google Cloud Storage Loader](/docs/pipeline-components-and-applications/loaders-storage-targets/google-cloud-storage-loader/index.md). 
 
-In addition, we recommend streaming the "failed events" Pub/Sub topic (containing data that fails to be processed successfully) to GCS, where it is possible to query the failed events to understand why they were not successfully processed and recover them using [Snowplow Event Recovery](https://github.com/snowplow-incubator/snowplow-event-recovery).
+We recommend streaming the "failed events" Pub/Sub topic (containing data that failed to be processed successfully) to GCS. From there, you may wish to load them into your destination of choice to query the failed events to understand why they were not successfully processed and recover them using [Snowplow Event Recovery](https://github.com/snowplow-incubator/snowplow-event-recovery).
```

Comment:
“query failed events” should link to the page about querying them in BigQuery (https://docs.snowplow.io/docs/managing-data-quality/failed-events/failed-events-in-athena-and-bigquery/)